studyBuddy/src/main/java/com/google/sps/models/authmodels/AuthInfo.java

```diff
@@ -0,0 +1,53 @@
+/* 
+This is class models information needed for a user to login and logout of to the site.
+
+Note: The Login Servlet returns a JSON response modeled like this class. 
+
+TODO: Add a method that generates the students's school from their email.       
```

Comment:
This could present an interesting issue that we may not need to solve. Since our main purpose for wanting the student's school is to only match them with students from their school, we don't actually have to know which school they go to, just if it is the same as someone else. This means that for simple functionality, we could just extract everything that comes after the '@' to fill a column in the table and add the same matching filter that we used for the subject.

If we do want to create some sort of look up that maps emails to schools, I think should do something very similar and use everything after the '@' as the keys. The question becomes how to add new entries into the map, primarily focused on who has the authority to add a mapping. It's unfeasible for the three of us to create a full map, but we cannot allow users to add mappings either.

Reply:
So if I understand we want use `@school.edu` as our key to search by ? Do we want users with '@gmail.com` extensions to also be able to get matches ?

Additionally,
**Since we are using google's User API, it is kind of hard to restrict people from signing in using their `gmail` account. (I will look into this but I am not sure it is possible)**

In the meantime here are some alternatives I have in mind:

- I can add some sort of restriction to users who have any extensions that is not `.edu` from getting any matches(and from being part of the results) and instead return an indicator in the response that the client can use to let the user know that they need to sign in with the school email in order to perform any searches.

- The client can then show a pop up or something on the page with this info for.

This way our site can only serve students. If you can think of a better approach let me know.

Ben